.github/workflows/build_slim_packages.yaml

```diff
@@ -52,8 +52,12 @@ jobs:
         path: ./rebar3.crashdump
     - name: packages test
       run: |
-        export CODE_PATH=$GITHUB_WORKSPACE
-        .ci/build_packages/tests.sh
+        set -euo pipefail
+        PKG_VSN="$(./pkg-vsn.sh)"
```

Comment:
The checkout action above is not using `fetch-depth: 0` to make a git clone, which `pkg-vsn.sh` depends on (it will fail on simple checkouts).

Although, the version is older than what I used recently (above is `v1`), so I'm not sure if it behaves in the same way.

Reply:
EMQ X release version is the of scheme: `VSN-COMMIT`
`VSN` is taken from `apps/emqx/include/emqx_release.hrl`
and `-COMMIT` is either empty or the commit hash of HEAD, it's empty when HEAD has no exact tag.
so it always works with shallow clone.

Reply:
I just commented because I encountered this issue [here](https://github.com/emqx/emqx/runs/4642834563?check_suite_focus=true#step:6:11), when I did not make a full clone.  If it works here, please ignore.  :smiley_cat: 

In my case, it failed on [`git rev-parse HEAD`](https://github.com/emqx/emqx/blob/b09d0d8e4c557011f2cc4f3500d231dfb02e6505/pkg-vsn.sh#L26) when it was not in a clone.

```sh
mkdir -p /tmp/vsn-test/apps/emqx/include

cp apps/emqx/include/emqx_release.hrl /tmp/vsn-test/apps/emqx/include/
cp pkg-vsn.sh /tmp/vsn-test/

cd /tmp/vsn-test/
./pkg-vsn.sh

# fatal: not a git repository (or any parent up to mount point /)
# Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
```